src/main/java/com/sem/btrouble/controller/Controller.java

```diff
@@ -289,24 +321,38 @@ public void removeObserver(LevelObserver observer) {
      */
     @Override
     public void notifyObserver() {
-        if(!Model.getCurrentRoom().hasBubbles()) {
-            for(LevelObserver levelObserver : levelObservers) {
+        if (!Model.getCurrentRoom().hasBubbles()) {
+            for (LevelObserver levelObserver : levelObservers) {
                 levelObserver.levelWon();
             }
         }
 
-        // Hardcoded player 1.
-        if(!Model.getPlayers().get(0).isAlive()) {
-            for(LevelObserver levelObserver : levelObservers) {
+        if (!anyLife()) {
+            for (LevelObserver levelObserver : levelObservers) {
                 levelObserver.levelLost();
             }
         }
 
-        if(getTimers().getLevelTimeLeft() <= 0) {
-            for(LevelObserver levelObserver : levelObservers) {
+        if (getTimers().getLevelTimeLeft() <= 0) {
+            for (LevelObserver levelObserver : levelObservers) {
                 levelObserver.levelLost();
             }
         }
 
     }
+    
+    /**
+     * Checks if there is a player in the room who is still alive.
+     * @return a boolean
+     */
+    public boolean anyLife() {
```

Comment:
Instead of using a boolean, when assigning alive = true you could immediately return true.
Return false on the last line. 
```
for (Player player : Model.getPlayers() {
    if (player.isAlive()) {
        return true;
    }
}
return false;
```